File: databases/mapdb_scala_shared_library/LibFunc.scala
import org.mapdb._
import java.util.concurrent.ConcurrentMap
//import org.graalvm.nativeimage.IsolateThread
//import org.graalvm.nativeimage.c.function.CEntryPoint

object LibFunc {
  
  def funcCall(): Unit = {
    // val db: DB = DBMaker.memoryDB().make()
    val db: DB = DBMaker.fileDB("database.db").make()
    val map: ConcurrentMap[String, String] = db.hashMap("map").make().asInstanceOf[ConcurrentMap[String, String]]

    val key = "google.com"
    var value = ""

    value = map.get(key)
    println(s"Got - Key: $key; Value: $value")

    val key2 = "bing.com"
    value = map.get(key2)
    println(s"Got - Key: $key2; Value: $value")

    db.close()
  }
}

File: databases/mapdb-mapdb-3.0.9/native-image/LibFunc.java
import org.mapdb.*;
import java.util.concurrent.ConcurrentMap;
import org.graalvm.nativeimage.IsolateThread;
import org.graalvm.nativeimage.c.function.CEntryPoint;

public class LibFunc {
	@CEntryPoint(name = "func_call")
	private static void funcCall( IsolateThread thread ) {
		//		DB db = DBMaker.memoryDB().make();
		DB db = DBMaker.fileDB("database.db").make();
		ConcurrentMap map = db.hashMap("map").make();

		String key   = "google.com" ;
		String value = "" ;
		/*
		   System.out.println( "Put - Key: " + key + "; Value: " + value );

		   map.put( key, value );
		   */

		key   = "google.com" ;

		value = (String) map.get( key );

		System.out.println( "Got - Key: " + key + "; Value: " + value );

		key   = "bing.com" ;

		value = (String) map.get( key );

		System.out.println( "Got - Key: " + key + "; Value: " + value );

		db.close();
	}
}

